### ISSUE TITLE
Proxyman cannot capture request, while Charles can

### ISSUE THREAD

**Issue by lemanhtien** [ProxymanApp/Proxyman#2192]
## Description
There is a case that Proxyman cannot capture the request, while Charles app can.
SSLProxy settings both set to "*"

## Steps to Reproduce

**Proxyman SSL settings**
![image](https://github.com/user-attachments/assets/444219c3-d026-4d2b-9c98-e3e67aba5232)
**Charles SSL settings**
<img width="574" alt="image" src="https://github.com/user-attachments/assets/1fd2cc9b-53bc-4529-afc6-8f1543df98fe">

## Expected Behavior

**Proxyman**
<img width="1191" alt="image" src="https://github.com/user-attachments/assets/64f8a37b-dea5-4048-9aa9-533b5d4e05d5">

**Charles**
<img width="1416" alt="image" src="https://github.com/user-attachments/assets/b57f6f99-915f-442f-9b4c-81a1d7ac6560">

## Environment
(Happened to all the Proxyman app version)
- App version: Proxyman 5.7.0
- macOS version: macOS 14.5

**Comment by NghiaTranUIT** [comment]
@lemanhtien it seems it's from iOS Simulators. May I ask if you installed the certificate on your iOS Simulator?

You can find this in the Certificate menu -> Install for iOS -> Simulator -> Follow all steps -> Try to make a request again

**Comment by NghiaTranUIT** [comment]
maybe It's from your app.

Let's try open Safari -> Visit www.google.com or www.producthunt.com -> Check if Proxyman can capture and decrypt these HTTPS data.

<img width="1929" alt="Screenshot 2024-11-20 at 1 29 12 PM" src="https://github.com/user-attachments/assets/996aa789-268c-4a41-adab-584502056d2d">

**Comment by lemanhtien** [comment]
@NghiaTranUIT I did not do a trivial mistake haha. All others request are fine (mean can be captured) on iOS simulator.
Just wondering why Charles can capture, but Proxyman cannot, with the same configuration.

**Comment by NghiaTranUIT** [comment]
so, Proxyman works fine after installing the certificate, doesn't it?

I guess you've already installed the Charles proxy certificate on your iOS Simulator for a long time, and forgot it. You can check in the Setting app -> General -> Trust Setting certificate -> I believe Charles's Cert is here.

**Comment by lemanhtien** [comment]
@NghiaTranUIT No, what I mean it doesn't work for some requests (the one above). Others requests are fine.

**Comment by NghiaTranUIT** [comment]
It's odd @lemanhtien. Does your HTTPS with IP have a special config? Does it called by URLSession or using 3rd-party network library?

I tested with my iOS app, make a REST request to `https://1.1.1.1` (IP) and it just works fine 🤔
